scripts/check_setting_visibility.py

```diff
@@ -0,0 +1,180 @@
+                        not_valid_setting_by_category[visibility_type].append(setting)
+
+                    # Add the 'basic' settings to the list
+                    if visibility_type == "basic":
+                        visible_settings_order.append(setting)
+
+
+        # Check whether the settings are added in right order or not.
+        # The basic settings should be in advanced, and advanced in expert
+        for visibility_type in setting_visibility:
+
+            # Skip the basic because it cannot be compared to previous list
+            if visibility_type == 'basic':
+                continue
+
+            all_settings_in_this_type = []
+            not_valid_setting_by_order[visibility_type] = []
+
+            item = setting_visibility_items[visibility_type]
+            for category, category_settings in item.items():
+                all_settings_in_this_type.extend(category_settings)
+
+
+            for setting in visible_settings_order:
+                if setting not in all_settings_in_this_type:
+                    not_valid_setting_by_order[visibility_type].append(setting)
+
+
+        # If any of the settings is defined not correctly then the test is failed
+        has_invalid_settings = False
+
+        for type, settings in not_valid_categories.items():
+            if len(settings) > 0:
+                has_invalid_settings = True
+                print("The following categories are defined incorrectly")
+                print("  Visibility type : '%s'" % (type))
+                print("  Incorrect categories : '%s'" % (settings))
+                print()
+
+
+
+        for type, settings in not_valid_setting_by_category.items():
+            if len(settings) > 0:
+                has_invalid_settings = True
+                print("The following settings do not exist anymore in fdmprinter definition or in wrong category")
+                print("  Visibility type : '%s'" % (type))
+                print("  Incorrect settings : '%s'" % (settings))
+                print()
+
+
+        for type, settings in not_valid_setting_by_order.items():
+            if len(settings) > 0:
+                has_invalid_settings = True
+                print("The following settings are defined in the incorrect order in setting visibility definitions")
+                print("  Visibility type : '%s'" % (type))
+                print("  Incorrect settings : '%s'" % (settings))
+
+        return has_invalid_settings
+
+
+if __name__ == "__main__":
+
+    all_setting_visibility_files = glob.glob(os.path.join(os.path.join(SCRIPT_DIR, "..", "resources", "setting_visibility"), '*.cfg'))
+    fdmprinter_def_path = os.path.join(SCRIPT_DIR, "..", "resources", "definitions", "fdmprinter.def.json")
+
+    inspector = SettingVisibilityInspection()
+    inspector.defineAllCuraSettings(fdmprinter_def_path)
+
+    setting_visibility_items = {}
+    for file_path in all_setting_visibility_files:
```

Comment:
Except the first file. Use `all_setting_visibility_files[1:]` after you've sorted these by weight.